test/hotspot/jtreg/compiler/vectorapi/TestVectorAddMulReduction.java

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2024, Arm Limited. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package compiler.vectorapi;
+
+import compiler.lib.ir_framework.*;
+
+import jdk.incubator.vector.DoubleVector;
+import jdk.incubator.vector.FloatVector;
+import jdk.incubator.vector.VectorOperators;
+import jdk.incubator.vector.VectorSpecies;
+
+import java.util.Random;
+
+import jdk.test.lib.Asserts;
+import jdk.test.lib.Utils;
+
+/**
+ * @test
+ * @bug 8320725
+ * @library /test/lib /
+ * @requires os.arch == "aarch64"
```

Comment:
I think there is no reason to only run the test on aarch64. We can run the test anywhere, but the applyIf specifies on what platforms the IR rules are executed.

Reply:
So you can use the `asimd` or `avx...` features for that.

Reply:
Yes, thanks. I will add the `applyIf...` rules instead. I added this line as I thought it would be easier to extend it to other platforms like x86 by including x86 in this line `@requires os.arch == "aarch64" || os.arch == "x86"` so that the @IR rules need not be changed or new ones need to be added but I guess your suggestion is better as the test can still be run on x86 (and not apply the IR rules instead).